### PR TITLE
front: fix layers button in editor

### LINKED ISSUE #6518: Layers are broken in editor
### What happened?

Whatever you choose in the layer card it is not displayed on the map in the editor. Only the tool you choose enable the display of layers:
image

### What did you expect to happen?

To be able to display any layer of my choice

### How can we reproduce it (as minimally and precisely as possible)?

.

### What operating system, browser and environment are you using?

- **Browser**: Firefox

### OSRD version (top right corner `Account` button > `Informations`)

dev 9346239

### PR DESCRIPTION
**Pull request by RomainValls**
closes #6518

### CODE REVIEW

**Review by Akctarus — APPROVED** [review]
Tested, LGTM

File: front/src/common/Map/Buttons/MapButtons.tsx [hunk 1]
```diff
@@ -58,11 +65,15 @@ export default function MapButtons({
...
-  editorState,
   isInEditor,
+  toolState,
+  setToolState,
 }: MapButtonsProps) {
   const dispatch = useDispatch();
   const { isOpen } = useContext(ModalContext);
+  const { openModal, editorState, activeTool } = useContext(
+    EditorContext
+  ) as ExtendedEditorContextType<unknown>;
```
Comment (clarani) [inline front/src/common/Map/Buttons/MapButtons.tsx]: We should not call the editorContext outside the editor, can you pass these const by props instead ? I think you can retrieve the openModal in the modalContext too (`const { isOpen, openModal } = useContext(ModalContext);`)
```suggestion
```

File: front/src/common/Map/Buttons/MapButtons.tsx [hunk 2]
```diff
@@ -121,7 +133,17 @@ export default function MapButtons({
...
+          <ButtonMapSettingsEditor
+            openModal={openModal}
+            editorState={editorState}
+            activeTool={activeTool}
+            toolState={toolState}
+            setToolState={setToolState}
+          />
+        ) : (
+          <ButtonMapSettings toggleMapSettings={() => toggleMapModal('SETTINGS')} />
+        )}
```
Comment (clarani) [inline front/src/common/Map/Buttons/MapButtons.tsx]: You don't need to create a new component ButtonMapSettingsEditor, you can just change the toggleMapSettings function :)

You should create a function toggleMapSettings like this:
```
  const openMapSettingsModal = useCallBack(() => {
    if (isInEditor) {
      if (activeTool && toolState && setToolState) {
        openModal(
          <LayersModal
            initialLayers={editorState.editorLayers}
            frozenLayers={activeTool.requiredLayers}
            selection={
              activeTool.id === 'select-items'
                ? (toolState as unknown as SelectionState).selection
                : undefined
            }
            onChange={({ newLayers }) => {
              if (activeTool.id === 'select-items') {
                const currentState = toolState as unknown as SelectionState;
                setToolState({
                  ...currentState,
                  selection: currentState.selection.filter((entity) =>
                    EDITOAST_TO_LAYER_DICT[entity.objType as EditoastType].every((layer) =>
                      newLayers.has(layer)
                    )
                  ),
                } as SelectionState);
              }
            }}
          />,
          'lg'
        );
      }
    } else {
      toggleMapModal('SETTINGS')
    }
  }, [])
```
and then you can just 
```suggestion
        <ButtonMapSettings toggleMapSettings={openMapSettingsModal} />
        
```

File: front/src/common/Map/Buttons/ButtonMapSettingsEditor.tsx
Comment (clarani) [inline front/src/common/Map/Buttons/ButtonMapSettingsEditor.tsx:1]: This file is not really needed, you can remove it :)

File: front/src/applications/editor/Editor.tsx
Comment (clarani) [inline front/src/applications/editor/Editor.tsx:1]: I would keep the editorState rather than tool state and setToolState

**Review by clarani — CHANGES_REQUESTED** [review]

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Review by clarani — APPROVED** [review]
LGTM, tested with Romain :white_check_mark:

**Pull request merged**